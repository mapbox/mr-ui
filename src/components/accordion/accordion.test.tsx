import React from 'react';
import Accordion from './accordion';

describe('Accordion', () => {
  /*
  let testCase;
  let wrapper;

  describe(testCases.basic.description, () => {
    beforeEach(() => {
      testCase = testCases.basic;
      wrapper = mount(React.createElement(testCase.component, testCase.props));
    });

    test('renders', () => {
      expect(
        shallow(React.createElement(testCase.component, testCase.props))
      ).toMatchSnapshot();
    });

    test('onToggle is called', () => {
      wrapper.find('button').first().simulate('click');
      expect(testCase.props.onToggle).toHaveBeenCalledTimes(1);
      expect(testCase.props.onToggle).toHaveBeenCalledWith('one');
    });

    test('activeItem prop updates active item', () => {
      wrapper.setProps({ activeItem: 'one' });
      wrapper.update();
      const item = wrapper.find('AccordionItem').first();
      expect(item.props()).toHaveProperty('active', true);
    });
  });

  describe(testCases.disabled.description, () => {
    beforeEach(() => {
      testCase = testCases.disabled;
      wrapper = mount(React.createElement(testCase.component, testCase.props));
    });

    test('renders', () => {
      expect(
        shallow(React.createElement(testCase.component, testCase.props))
      ).toMatchSnapshot();
    });

    test('disabled works', () => {
      const props = wrapper.find('button').last().props();
      expect(props).toHaveProperty('disabled', true);
    });
  });

  describe(testCases.allOptions.description, () => {
    beforeEach(() => {
      testCase = testCases.allOptions;
      wrapper = shallow(
        React.createElement(testCase.component, testCase.props)
      );
    });

    test('renders', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('extra props works', () => {
      expect(testCase.props).toHaveProperty('onToggle');
      expect(testCase.props).toHaveProperty('activeItem');

      // Optional props passed to items
      const props = wrapper.find('AccordionItem').first().props();
      expect(props).toHaveProperty('active', true);
      expect(props).toHaveProperty('themeItem');
      expect(props).toHaveProperty('themeItemHeader');
      expect(props).toHaveProperty('themeItemBody');
    });
  });
*/
});

/*
testCases.basic = {
  description: 'basic',
  component: ControlledAccordion,
  props: {
    onToggle: safeSpy(),
    items: [
      { id: 'one', header: () => <span>one</span>, body: 'First body' },
      { id: 'two', header: () => <span>two</span>, body: 'Second body' },
      { id: 'three', header: () => <span>three</span>, body: 'Third body' }
    ]
  }
};

testCases.disabled = {
  description: 'disabled',
  component: ControlledAccordion,
  props: {
    onToggle: safeSpy(),
    items: [
      { id: 'one', header: () => <span>one</span>, body: 'First body' },
      { id: 'two', header: () => <span>two</span>, body: 'Second body' },
      {
        id: 'three',
        header: () => <span>three</span>,
        body: 'Third body',
        disabled: true
      }
    ]
  }
};

testCases.allOptions = {
  description: 'all options',
  component: ControlledAccordion,
  props: {
    onToggle: safeSpy(),
    items: [
      {
        id: 'one',
        header: (active) => {
          return (
            <div>
              one
              {active && <span className="ml6 txt-normal">I am active</span>}
            </div>
          );
        },
        body: 'First body'
      },
      {
        id: 'two',
        header: (active) => {
          return (
            <div>
              two
              {active && <span className="ml6 txt-normal">I am active</span>}
            </div>
          );
        },
        body: 'Second body'
      },
      {
        id: 'third',
        header: (active) => {
          return (
            <div>
              third
              {active && <span className="ml6 txt-normal">I am active</span>}
            </div>
          );
        },
        body: 'Third body'
      }
    ],
    activeItem: 'one',
    themeItem: 'bg-red-faint border-b border--red-dark',
    themeItemHeader:
      'bg-red txt-l txt-bold link link--white bg-red-dark-on-hover px6 py6',
    themeItemBody: 'py6 px6 txt-l bg-red-light'
  }
};
*/
