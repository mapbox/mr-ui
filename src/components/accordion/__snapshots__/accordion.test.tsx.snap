// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion all options renders 1`] = `
<body>
  <div>
    <div>
      <div
        class="bg-red-faint border-b border--red-dark"
        data-state="open"
      >
        <button
          aria-controls="radix-24"
          aria-expanded="true"
          class="w-full bg-red txt-l txt-bold link link--white bg-red-dark-on-hover px6 py6 cursor-pointer"
          data-radix-collection-item=""
          data-state="open"
          data-testid="accordion-trigger-one"
          id="radix-25"
          type="button"
        >
          <div>
            one
            <span
              class="ml6 txt-normal"
            >
              I am active
            </span>
          </div>
        </button>
        <div
          aria-labelledby="radix-25"
          data-state="open"
          id="radix-24"
          role="region"
          style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width); transition-duration: 0s; animation-duration: 0s; animation-fill-mode: none;"
        >
          <div
            class="py6 px6 txt-l bg-red-light"
          >
            First body
          </div>
        </div>
      </div>
      <div
        class="bg-red-faint border-b border--red-dark"
        data-state="closed"
      >
        <button
          aria-controls="radix-26"
          aria-expanded="false"
          class="w-full bg-red txt-l txt-bold link link--white bg-red-dark-on-hover px6 py6 cursor-pointer"
          data-radix-collection-item=""
          data-state="closed"
          data-testid="accordion-trigger-two"
          id="radix-27"
          type="button"
        >
          <div>
            two
          </div>
        </button>
        <div
          aria-labelledby="radix-27"
          data-state="closed"
          hidden=""
          id="radix-26"
          role="region"
          style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
        />
      </div>
      <div
        class="bg-red-faint border-b border--red-dark"
        data-state="closed"
      >
        <button
          aria-controls="radix-28"
          aria-expanded="false"
          class="w-full bg-red txt-l txt-bold link link--white bg-red-dark-on-hover px6 py6 cursor-pointer"
          data-radix-collection-item=""
          data-state="closed"
          data-testid="accordion-trigger-third"
          id="radix-29"
          type="button"
        >
          <div>
            third
          </div>
        </button>
        <div
          aria-labelledby="radix-29"
          data-state="closed"
          hidden=""
          id="radix-28"
          role="region"
          style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
        />
      </div>
    </div>
  </div>
</body>
`;

exports[`Accordion basic renders 1`] = `
<body>
  <div>
    <div>
      <div
        data-state="closed"
      >
        <button
          aria-controls="radix-0"
          aria-expanded="false"
          class="w-full txt-s txt-bold txt-truncate link link--gray py6 cursor-pointer"
          data-radix-collection-item=""
          data-state="closed"
          data-testid="accordion-trigger-one"
          id="radix-1"
          type="button"
        >
          <span>
            one
          </span>
        </button>
        <div
          aria-labelledby="radix-1"
          data-state="closed"
          hidden=""
          id="radix-0"
          role="region"
          style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
        />
      </div>
      <div
        data-state="closed"
      >
        <button
          aria-controls="radix-2"
          aria-expanded="false"
          class="w-full txt-s txt-bold txt-truncate link link--gray py6 cursor-pointer"
          data-radix-collection-item=""
          data-state="closed"
          data-testid="accordion-trigger-two"
          id="radix-3"
          type="button"
        >
          <span>
            two
          </span>
        </button>
        <div
          aria-labelledby="radix-3"
          data-state="closed"
          hidden=""
          id="radix-2"
          role="region"
          style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
        />
      </div>
      <div
        data-state="closed"
      >
        <button
          aria-controls="radix-4"
          aria-expanded="false"
          class="w-full txt-s txt-bold txt-truncate link link--gray py6 cursor-pointer"
          data-radix-collection-item=""
          data-state="closed"
          data-testid="accordion-trigger-three"
          id="radix-5"
          type="button"
        >
          <span>
            three
          </span>
        </button>
        <div
          aria-labelledby="radix-5"
          data-state="closed"
          hidden=""
          id="radix-4"
          role="region"
          style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
        />
      </div>
    </div>
  </div>
</body>
`;

exports[`Accordion disabled renders 1`] = `
<body>
  <div>
    <div>
      <div
        data-state="closed"
      >
        <button
          aria-controls="radix-12"
          aria-expanded="false"
          class="w-full txt-s txt-bold txt-truncate link link--gray py6 cursor-pointer"
          data-radix-collection-item=""
          data-state="closed"
          data-testid="accordion-trigger-one"
          id="radix-13"
          type="button"
        >
          <span>
            one
          </span>
        </button>
        <div
          aria-labelledby="radix-13"
          data-state="closed"
          hidden=""
          id="radix-12"
          role="region"
          style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
        />
      </div>
      <div
        data-state="closed"
      >
        <button
          aria-controls="radix-14"
          aria-expanded="false"
          class="w-full txt-s txt-bold txt-truncate link link--gray py6 cursor-pointer"
          data-radix-collection-item=""
          data-state="closed"
          data-testid="accordion-trigger-two"
          id="radix-15"
          type="button"
        >
          <span>
            two
          </span>
        </button>
        <div
          aria-labelledby="radix-15"
          data-state="closed"
          hidden=""
          id="radix-14"
          role="region"
          style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
        />
      </div>
      <div
        data-state="closed"
      >
        <button
          aria-controls="radix-16"
          aria-expanded="false"
          class="w-full txt-s txt-bold txt-truncate link link--gray py6 cursor-default color-gray-light"
          data-radix-collection-item=""
          data-state="closed"
          data-testid="accordion-trigger-three"
          disabled=""
          id="radix-17"
          type="button"
        >
          <span>
            three
          </span>
        </button>
        <div
          aria-labelledby="radix-17"
          data-state="closed"
          hidden=""
          id="radix-16"
          role="region"
          style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
        />
      </div>
    </div>
  </div>
</body>
`;
